shuup/admin/modules/orders/views/detail.py

```diff
@@ -68,7 +68,7 @@ def post(self, request, *args, **kwargs):
             raise Problem(_("Unable to set order as completed at this point."))
         if new_status.role == OrderStatusRole.CANCELED and not order.can_set_canceled():
             raise Problem(_("You can't cancel orders that are paid, shipped, or already canceled."))
-        order.status = new_status
+        order.change_status(next_status=new_status, user=request.user)
         order.save(update_fields=("status",))
```

Comment:
I think it would be better to move the `order.save()` to the new `change_status` method.

Reply:
Perfect.